WordPress/gutenberg
# [Mobile] - Fix missing custom gradient indicator in the color palette

Fixes https://github.com/WordPress/gutenberg/issues/42017

## What?
This PR fixes a regression most likely introduced in https://github.com/WordPress/gutenberg/pull/38417 where we added support for multiple color palettes where the custom color indicator is not shown for custom gradient colors.

## Why?
To fix a UI regression and possibly avoid confusion for users when they customize their gradient colors.

## How?
The custom indicator wasn't being shown due to not having all of the gradient colors available for the current color segment being rendered.

We also limit the custom indicator to only be visible in the first available color palette row, since users can have up to three: `Default`, `Theme`, and `Custom` colors.

## Testing Instructions

> [!NOTE]  
> These changes need to be tested using the main host apps and a theme that has custom gradients e.g: [Twenty Twenty-Four](https://wordpress.org/themes/twentytwentyfour/)

- Add a Buttons block.
- Open the settings for the Button block.
- Tap Background.
- Tap Gradient.
- Select a preset gradient.
- Tap Customize Gradient.
- Modify the gradient.
- Tap Back
- Expect to see the Custom indicator at the end of the list
- Tap Back again
- Tap Background
- Expect to see the list to auto-scroll to the end where the Custom indicator should be visible

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/4885740/66d90cdb-f11c-4590-a932-0d266de36683
